apps/src/lib/node/ledger/ethereum_node/mod.rs

```diff
@@ -196,7 +231,21 @@ pub mod eth_fullnode {
     }
 }
 
-#[cfg(feature = "eth-fullnode")]
-pub use eth_fullnode::EthereumNode;
-#[cfg(not(feature = "eth-fullnode"))]
-pub use test_tools::mock_eth_fullnode::EthereumNode;
+/// Starts an Ethereum fullnode in a subprocess and returns a handle for
+/// monitoring it using [`monitor`], as well as a channel for halting it.
+pub async fn start(url: &str, real: bool) -> Result<(Subprocess, Sender<()>)> {
+    if real {
+        let (node, sender) = eth_fullnode::EthereumNode::new(url).await?;
+        Ok((Subprocess::Geth(node), sender))
+    } else {
+        let (node, sender) =
+            test_tools::mock_eth_fullnode::EthereumNode::new().await?;
+        Ok((Subprocess::Mock(node), sender))
+    }
+}
+
+/// Represents a subprocess running an Ethereum full node
+pub enum Subprocess {
+    Mock(test_tools::mock_eth_fullnode::EthereumNode),
+    Geth(eth_fullnode::EthereumNode),
+}
```

Comment:
Having this enum is unfortunate. I tried to make both `EthereumNode` types implement the same (async) trait but ran into issues with futures not being sendable across threads. I think the best refactor to do later on would be to change `run_aux` so that if `ledger.ethereum.geth_startup = false`, we don't start any sort of mock node at all and then we could remove it altogether - but at the moment `run_aux` is sort of hardcoded in our branch to expect five handles to have to manage.